__std__/nix/toolchain/packages/pre-commit-check.nix

```diff
@@ -0,0 +1,40 @@
+{ inputs, cell }:
+
+cell.packages.todo-derivation
+
+# TODO(std) we need stylish-haskell for this
+
+# # Configure project pre-commit hooks
+# pre-commit-check = nix-pre-commit-hooks.run {
```

Comment:
I have to say that I have a little bit of an opinion _against_ `nix-pre-commit`. I think it's a bad idea.

Pre commit tries to be a package manager. Nix _is_ a package manager.

In the past (quite a while ago), I've been myself contributing to `nix-pre-commit` when I realized how wired of an idea it is to use pre-commit just to go our of the way to disable (at that times: sometimes hacky) it's package manager capabilities.

I even went upstream to make that bypassing a little more elegant and was bluntly rejected by incredulity on how one could not want it as a package manager.

I then just started to use bash scripts + Nix that would have been put in their place via Numtide's devshell git hook module.

Yet, at another turn of my commit hook journey, I was pointed to `lefthook`.

I think that's actually q really good solution, because it focuses on what it does best (without trying to be a package manager), does some interesting parallelization and in general is well maintained with a solid design.

Hence, `lefthook` already ships as a Nixago Pebble within `std`, as of fairly recently.